src/window/floatingwindow.cpp

```diff
@@ -44,6 +47,65 @@ void ProxyFloatingWindow::onMaximumSizeChanged() {
 	emit this->maximumSizeChanged();
 }
 
+QObject* ProxyFloatingWindow::parentWindow() const { return this->mParentWindow; }
+
+void ProxyFloatingWindow::setParentWindow(QObject* window) {
+	if (window == this->mParentWindow) return;
+
+	if (this->mParentWindow) {
+		QObject::disconnect(this->mParentWindow, nullptr, this, nullptr);
+		QObject::disconnect(this->mParentProxyWindow, nullptr, this, nullptr);
+	}
+
+	if (window) {
+		if (auto* proxy = qobject_cast<ProxyWindowBase*>(window)) {
+			this->mParentProxyWindow = proxy;
+		} else if (auto* interface = qobject_cast<WindowInterface*>(window)) {
+			this->mParentProxyWindow = interface->proxyWindow();
+		} else {
+			qmlWarning(this) << "parentWindow must be a quickshell window.";
+			return;
+		}
+
+		this->mParentWindow = window;
+		QObject::connect(
+		    this->mParentWindow,
+		    &QObject::destroyed,
+		    this,
+		    &ProxyFloatingWindow::onParentDestroyed
+		);
+
+		QObject::connect(
+		    this->mParentProxyWindow,
+		    &ProxyWindowBase::backingWindowChanged,
+		    this,
+		    &ProxyFloatingWindow::updateTransientParent
+		);
+	} else {
+		this->mParentWindow = nullptr;
+		this->mParentProxyWindow = nullptr;
+	}
+
+	this->updateTransientParent();
+	emit this->parentWindowChanged();
+}
+
+void ProxyFloatingWindow::onParentDestroyed() {
+	this->mParentWindow = nullptr;
+	this->mParentProxyWindow = nullptr;
+	this->updateTransientParent();
+	emit this->parentWindowChanged();
+}
+
+void ProxyFloatingWindow::updateTransientParent() {
+	if (!this->window) return;
+
+	auto* parentBacking =
+	    this->mParentProxyWindow ? this->mParentProxyWindow->backingWindow() : nullptr;
+
+	this->window->setTransientParent(parentBacking);
+}
```

Comment:
This implementation will not work unless the window objects have already been created, but it should work regardless. The hook point you want is visibility change, and its likely not possible to even change it after the window has been made visible.

Reply:
It's not possible to change it after creation, I simplified it to just set it in postComplete, and annotated it in the documentation that it cant be changed